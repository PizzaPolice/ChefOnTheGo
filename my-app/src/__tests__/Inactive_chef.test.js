import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import InactiveChef from '../components/inactive_chef.jsx'


it('InactiveChef component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<InactiveChef />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

// This test is shallow rendering our InactiveChef component
describe('InactiveChef Component', () => {
    it('InactiveChef should render without throwing an error', () => {
        expect(shallow(<InactiveChef />).exists(<form className='InactiveChef'></form>)).toBe(true)
      })
      it("should render initial layout", () => {
        const component = shallow(<InactiveChef />);
        expect(component.getElements()).toMatchSnapshot();
        })
    });