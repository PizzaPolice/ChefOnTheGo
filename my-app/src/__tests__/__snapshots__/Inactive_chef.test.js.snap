// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InactiveChef Component should render initial layout 1`] = `
Array [
  <div>
    <title>
        Inactive Chef
    </title>
    <h2>
        You are currently inactive
    </h2>
    <form
        className="input"
        onSubmit={[Function]}
    >
        <h3>
            How long do you want to be active until
        </h3>
        <input
            defaultValue="0"
            name="hour"
            onChange={[Function]}
            type="number"
        />
        <b>
            Hours
        </b>
        <input
            defaultValue="0"
            name="minute"
            onChange={[Function]}
            type="number"
        />
        <b>
            Minutes
        </b>
        <input
            type="submit"
            value="Start Serving!"
        />
    </form>
    <h3>
        Or change your offered menu
    </h3>
    <Link
        replace={false}
        to="/edit_dishes"
    >
        <button>
            Edit Dishes
        </button>
    </Link>
</div>,
]
`;
